src/Symfony/Component/Yaml/Inline.php

```diff
@@ -270,97 +266,143 @@ private static function dumpArray($value, $flags)
         return sprintf('{ %s }', implode(', ', $output));
     }
 
-    /**
-     * Parses a scalar to a YAML string.
-     *
-     * @param string $scalar
-     * @param int    $flags
-     * @param string $delimiters
-     * @param array  $stringDelimiters
-     * @param int    &$i
-     * @param bool   $evaluate
-     * @param array  $references
-     *
-     * @return string A YAML string
-     *
-     * @throws ParseException When malformed inline YAML string is parsed
-     *
-     * @internal
-     */
-    public static function parseScalar($scalar, $flags = 0, $delimiters = null, $stringDelimiters = array('"', "'"), &$i = 0, $evaluate = true, $references = array())
+    private static function parseValue(StringReader $reader, $flags = 0, $delimiters = null, $evaluate = true, $references = array())
     {
-        if (in_array($scalar[$i], $stringDelimiters)) {
-            // quoted scalar
-            $output = self::parseQuotedScalar($scalar, $i);
-
-            if (null !== $delimiters) {
-                $tmp = ltrim(substr($scalar, $i), ' ');
-                if (!in_array($tmp[0], $delimiters)) {
-                    throw new ParseException(sprintf('Unexpected characters (%s).', substr($scalar, $i)));
+        $reader->consumeWhiteSpace();
+        if ('!' === $reader->peek()) {
+            if ($reader->readString('!str')) {
+                $reader->consumeWhiteSpace();
+
+                return self::parseScalar($reader, $flags, $delimiters, false, $references);
+            }
+            // Non-specific tag
+            if ($reader->readString('! ')) {
+                $value = self::parseValueInner($reader, $flags, $delimiters, $evaluate, $references);
+                if (is_scalar($value)) {
+                    // @todo deprecate the int conversion as it
+                    // should be a string
+                    // @see http://www.yaml.org/spec/1.2/spec.html#tag/non-specific/
+                    return (int) $value;
```

Comment:
Should we consider that as a bug? It looks like it is due to a misunderstanding of the specification and its normal behavior is very useful as `{false: foo}` should not be allowed in php (`false` can't be a key), it only works because symfony's parser consider it as a string but it shouldn't. `! false` or `!!str false` should be used instead. 
As I would like to deprecate using special scalars without explicitly casting them to a string, it could be very useful :)